src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -505,14 +500,14 @@ public Query termQuery(Object value, @Nullable QueryParseContext context) {
 
     @Override
     public Filter termFilter(Object value, @Nullable QueryParseContext context) {
-        return new TermFilter(names().createIndexNameTerm(indexedValueForSearch(value)));
+        return Queries.wrap(new TermQuery(names().createIndexNameTerm(indexedValueForSearch(value))));
     }
 
     @Override
     public Filter termsFilter(List values, @Nullable QueryParseContext context) {
         switch (values.size()) {
         case 0:
-            return new MatchNoDocsFilter();
+            return Queries.MATCH_NO_FILTER;
```

Comment:
Do we really need singletons here? I'm worried about boosts getting changed somewhere. It seems completely unnecessary when we have a method to create a new one.